### PR TITLE
Implement "hash_entry" (make_hash)

### PR DESCRIPTION
**Pull request by Connoropolous**
Please focus not on my matches code, but on lines 15, 16, 17 that actually define a spec for hash_entry(entry) and what it returns

The proposed dev camp spec
https://github.com/holochain/hdk-rust/projects/1?fullscreen=true

https://github.com/holochain/hdk-rust/issues/29

### CODE REVIEW

File: zomes/blog/code/src/main_capability.rs [hunk 1]
```diff
@@ -3,6 +3,25 @@ use hdk::globals::G_MEM_STACK;
...
+        match hdk::get_entry(post_hash) {
+            Err(hdk_error) => hdk_error.to_json(),
+            Ok(maybe_entry) => match maybe_entry {
+                None => json!({}),
+                Some(entry) => match serde_json::from_str(&entry) {
+                    Err(err) => json!({"error deserializing post": err.to_string()}),
+                    Ok(post) => {
+                        // Post is an "entry", so give it to get_entry_address
+                        let maybe_address = hdk::hash_entry(post);
+                        match maybe_address {
```
Comment (AshantiMutinta) [inline zomes/blog/code/src/main_capability.rs]: I would reorder some of these with an ok, then Err. Just makes it easier to read imo.

Also part of me thinks we can churn down some of the code using an and_then clause but definitely not needed.

Comment (ddd-mtl) [reply zomes/blog/code/src/main_capability.rs]: Connor made the code according to [this discussion](https://github.com/holochain/app-spec-rust/pull/20).
Nico said he didn't find a better way with combinators.

Comment (Connoropolous) [reply zomes/blog/code/src/main_capability.rs]: yeah hehehe. Personally I am loosely interested in the discussion of rust syntax stuffs, but verry interested in the specific API for `hash_entry`

**Comment by ddd-mtl** [comment]
This has shown me how this "spec as an app" thing isn't suffisant for spec'ing.
The problem is when doing just a simple app, you are not covering all the edge-cases of the API, we are missing out on at least half the work of spec'ing.
This PR doesn't give me any clue what the function `make_entry()` is for and what errors/panics it might return.
The function `get_an_address()` doesn't seem to make any sense as you give the hash as input and expect the same hash as output. It could make sense as a testing function but not for an actual app.

I find "spec as an app" useful for figuring out a clear API with clear names, and for helping us see any blindspots. But those blindspots are already reported by all the other dev teams making hApps, like for example the entry type namespace problem.

**Comment by zippy** [comment]
> The function `get_an_address()` doesn't seem to make any sense as you give the hash as input and expect the same hash as output. It could make sense as a testing function but not for an actual app.

I agree with this one.  I've added a more natural function `hash_post()` for the test.

**Pull request merged**